services/workflows-service/src/secrets-manager/aws-secrets-manager.ts

```diff
@@ -0,0 +1,107 @@
+import { SecretsManager } from '@/secrets-manager/secrets-manager';
+import {
+  CreateSecretCommand,
+  GetSecretValueCommand,
+  PutSecretValueCommand,
+  ResourceNotFoundException,
+  SecretsManagerClient,
+} from '@aws-sdk/client-secrets-manager';
+import { z } from 'zod';
+
+const SecretStringSchema = z
+  .string()
+  .transform((value, ctx) => {
+    try {
+      return JSON.parse(value);
+    } catch (error) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'Secret value is not a valid JSON',
+      });
+
+      return z.NEVER;
+    }
+  })
+  .pipe(z.record(z.string(), z.string()));
+
+export class AwsSecretsManager implements SecretsManager {
+  private client: SecretsManagerClient;
+  private prefix: string;
+  private customerId: string;
+
+  constructor({
+    prefix,
+    customerId,
+    awsRegion,
+    awsAccessKeyId,
+    awsSecretAccessKey,
+  }: {
+    prefix: string;
+    customerId: string;
+    awsRegion: string;
+    awsAccessKeyId: string;
+    awsSecretAccessKey: string;
+  }) {
+    this.client = new SecretsManagerClient({
+      region: awsRegion,
+      credentials: {
+        accessKeyId: awsAccessKeyId,
+        secretAccessKey: awsSecretAccessKey,
+      },
+    });
+    this.prefix = prefix;
+    this.customerId = customerId;
+  }
+
+  async getAll() {
+    let secretString;
+
+    try {
+      const result = await this.client.send(
+        new GetSecretValueCommand({
+          SecretId: this.getSecretName(),
+        }),
+      );
+
+      secretString = result.SecretString;
+    } catch (error) {
+      if (error instanceof ResourceNotFoundException) {
+        // Create secret lazily, only if it doesn't exist
+        await this.createSecret();
+
+        return {};
+      }
+
+      throw error;
+    }
+
+    return SecretStringSchema.parse(secretString);
+  }
+
+  async set(data) {
+    const dataToSet = {
+      ...(await this.getAll()),
+      ...data,
+    };
+
+    await this.client.send(
+      new PutSecretValueCommand({
+        SecretId: this.getSecretName(),
+        SecretString: JSON.stringify(dataToSet),
+      }),
+    );
+  }
+
+  private async createSecret() {
+    await this.client.send(
+      new CreateSecretCommand({
+        Name: this.getSecretName(),
+        SecretString: JSON.stringify({}),
+      }),
+    );
+  }
+
+  private getSecretName() {
+    return `${this.prefix}/${this.customerId}`;
```

Comment:
In addition to this change: https://github.com/ballerine-io/ballerine/pull/2569/files#r1694232169

you should also change to something like : `/${env.ENV_NAME}/${this.prefix}/${}${this.customerId}`

Reply:
Why is it better?